src/Logging/src/Abstractions/MessageProcessingLogger.cs

```diff
@@ -46,31 +46,32 @@ public bool IsEnabled(LogLevel logLevel)
 
     public virtual void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Exception exception, Func<TState, Exception, string> formatter)
     {
+        ArgumentGuard.NotNull(formatter);
+
         if (!IsEnabled(logLevel))
         {
             return;
         }
 
-        ArgumentGuard.NotNull(formatter);
+        Func<TState, Exception, string> compositeFormatter = (innerState, innerException) =>
+            ApplyMessageProcessors(innerState, innerException, formatter, MessageProcessors);
 
+        Delegate.Log(logLevel, eventId, state, exception, compositeFormatter);
+    }
+
+    private static string ApplyMessageProcessors<TState>(TState state, Exception exception, Func<TState, Exception, string> formatter,
+        IEnumerable<IDynamicMessageProcessor> processors)
+    {
         string message = formatter(state, exception);
 
-        if (MessageProcessors != null)
+        if (processors != null)
         {
-            foreach (IDynamicMessageProcessor processor in MessageProcessors)
+            foreach (IDynamicMessageProcessor processor in processors)
             {
                 message = processor.Process(message);
             }
         }
 
-        if (!string.IsNullOrEmpty(message) || exception != null)
```

Comment:
This  null/empty message or exception was removed- was this intentional?

Reply:
Yes. The logic has moved into the formatters, for example: https://github.com/dotnet/runtime/blob/main/src/libraries/Microsoft.Extensions.Logging.Console/src/SimpleConsoleFormatter.cs#L50-L53